build fails with gcc 4.8

The CI workflow passes the recent PRs except for gcc 4.8 on Ubuntu 20.

Here is what the CI says

```
/home/runner/work/_temp/Library/cpp11/include/cpp11/protect.hpp:161:31: warning: format not a string literal and no format arguments [-Wformat-security]
     decltype(std::declval<F*>()(std::declval<A&&>()...)) operator()(A&&... a) const {
                               ^
/home/runner/work/_temp/Library/cpp11/include/cpp11/protect.hpp: In instantiation of ‘struct cpp11::detail::closure<void(SEXPREC*, const char*, ...), SEXPREC*&, const char*&&>’:
/home/runner/work/_temp/Library/cpp11/include/cpp11/protect.hpp:185:89:   required from ‘void cpp11::protect::noreturn_function<F>::operator()(A&& ...) const [with A = {SEXPREC*&, const char*}; F = void(SEXPREC*, const char*, ...)]’
/home/runner/work/_temp/Library/cpp11/include/cpp11/protect.hpp:234:29:   required from ‘void cpp11::stop(const string&, Args ...) [with Args = {}; std::string = std::basic_string<char>]’
errors.cpp:5:84:   required from here
/home/runner/work/_temp/Library/cpp11/include/cpp11/protect.hpp:148:29: warning: format not a string literal and no format arguments [-Wformat-security]
   decltype(std::declval<F*>()(std::declval<Aref>()...)) operator()() && {
                             ^
‘
Internal compiler error: Error reporting routines re-entered.
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
Preprocessed source stored into /tmp/ccOS2WLf.out file, please attach this to your bugreport.
make: *** [/opt/R/4.4.0/lib/R/etc/Makeconf:204: errors.o] Error 1
```

I tried to replicate this locally with VirtualBox, and so far I didn't succeed.

```
Enter one or more numbers, or an empty line to skip updates: 
Error: Could not find tools necessary to compile a package
Call `pkgbuild::check_build_tools(debug = TRUE)` to diagnose the problem.
```

--- a/.github/workflows/R-CMD-check.yaml
+++ b/.github/workflows/R-CMD-check.yaml
@@ -36,7 +36,6 @@ jobs:
           - {os: ubuntu-20.04,   r: 'devel', http-user-agent: 'release'}
           - {os: ubuntu-20.04,   r: 'release'}
           - {os: ubuntu-20.04,   r: 'release', custom: 'no-cpp11test'}
-          - {os: ubuntu-20.04,   r: 'release', custom: 'gcc 4.8'}
           - {os: ubuntu-20.04,   r: 'oldrel-1'}
           - {os: ubuntu-20.04,   r: 'oldrel-2'}
           - {os: ubuntu-20.04,   r: 'oldrel-3'}
@@ -56,24 +55,6 @@ jobs:
           r-version: ${{ matrix.config.r }}
           http-user-agent: ${{ matrix.config.http-user-agent }}
           use-public-rspm: true
-
-      - name : Install compiler version
-        # We check on this old compiler specifically to support CentOS 7,
-        # which uses this gcc version. RStudio products support CentOS 7 through
-        # June 2024.
-        # https://github.com/r-lib/cpp11/pull/78
-        # https://www.rstudio.com/about/platform-support/
-        # Ubuntu 20.04 technically dropped support for gcc 4.8, so we have to
-        # add old archives back in manually to install it
-        # https://github.com/r-lib/cpp11/pull/279
-        if: matrix.config.custom == 'gcc 4.8'
-        run: |
-          echo "deb http://dk.archive.ubuntu.com/ubuntu/ xenial main" | sudo tee -a /etc/apt/sources.list
-          echo "deb http://dk.archive.ubuntu.com/ubuntu/ xenial universe" | sudo tee -a /etc/apt/sources.list
-          sudo apt update
-          sudo apt-get install -y g++-4.8
-          mkdir ~/.R/
-          echo $'CXX1X=g++-4.8\nCXX11=g++-4.8\nCC=gcc-4.8 -std=c99\nCXX=g++-4.8 -std=gnu++11' >> ~/.R/Makevars
 
       - name: Install macOS system dependencies
         if: runner.os == 'macOS'

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,5 +1,10 @@
 # cpp11 (development version)
 
+* Dropped support for gcc 4.8, mainly an issue for extremely old CentOS 7
+  systems which used that as their default compiler. As of June 2024, CentOS 7
+  is past its Vendor end of support date and therefore also out of scope for
+  Posit at this time (#359).
+
 # cpp11 0.4.7
 
 * Internal changes requested by CRAN to fix invalid format string tokens

